pkg/rke2/rke2_linux.go

```diff
@@ -62,9 +65,12 @@ func initExecutor(clx *cli.Context, cfg Config, dataDir string, disableETCD bool
 			Path: cfg.CloudProviderConfig,
 		}
 		if clx.String("node-name") == "" && cfg.CloudProviderName == "aws" {
-			fqdn, err := hostnameFQDN()
+			fqdn, err := hostnameFromMetadataEndpoint()
 			if err != nil {
-				return nil, err
+				fqdn, err = hostnameFQDN()
```

Comment:
What happens to this error value?

Reply:
If we're logging relevant errors in the function itself and what gets returned is less meaningful and can be ignore, remove the error from the return and check here for an empty string, and do the additional call to `hostnameFQDN`.

Reply:
It gets return below if it is non-nil.

Reply:
Implicit reuse of err vals can lead to problematic code. It's generally best to `:=` your errors.

Reply:
Updated.